payroll_account/models/hr_payroll_account.py

```diff
@@ -0,0 +1,322 @@
+    )
+    journal_id = fields.Many2one(
+        "account.journal",
+        "Salary Journal",
+        readonly=True,
+        required=True,
+        default=lambda self: self.env["account.journal"].search(
+            [("type", "=", "general")], limit=1
+        ),
+    )
+    move_id = fields.Many2one(
+        "account.move", "Accounting Entry", readonly=True, copy=False
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        if "journal_id" in self.env.context:
+            for vals in vals_list:
+                vals["journal_id"] = self.env.context.get("journal_id")
+        return super().create(vals_list)
+
+    @api.onchange("contract_id")
+    def onchange_contract(self):
+        res = super().onchange_contract()
+        self.journal_id = (
+            self.contract_id.journal_id.id
+            or (not self.contract_id and self.default_get(["journal_id"])["journal_id"])
+            or self.journal_id
+        )
+        return res
+
+    def action_payslip_cancel(self):
+        for payslip in self:
+            if not payslip.move_id.journal_id.restrict_mode_hash_table:
+                payslip.move_id.with_context(force_delete=True).button_cancel()
+                payslip.move_id.with_context(force_delete=True).unlink()
+            else:
+                payslip.move_id._reverse_moves()
+                payslip.move_id = False
+        return super().action_payslip_cancel()
+
+    def action_payslip_done(self):
+        res = super().action_payslip_done()
+
+        for slip in self:
+            line_ids = []
+            debit_sum = 0.0
+            credit_sum = 0.0
+            date = slip.date or slip.date_to
+            currency = (
+                slip.company_id.currency_id or slip.journal_id.company_id.currency_id
+            )
+
+            name = _("Payslip of %s") % (slip.employee_id.name)
+            move_dict = {
+                "narration": name,
+                "ref": slip.number,
+                "journal_id": slip.journal_id.id,
+                "date": date,
+            }
+            for line in slip.line_ids:
+                amount = currency.round(slip.credit_note and -line.total or line.total)
+                if currency.is_zero(amount):
+                    continue
+                debit_account_id = line.salary_rule_id.account_debit.id
+                credit_account_id = line.salary_rule_id.account_credit.id
+                account_id = debit_account_id or credit_account_id
+                analytic_salary_id = line.salary_rule_id.analytic_account_id.id
+                move_line_analytic_ids = []
+                if analytic_salary_id:
+                    move_line_analytic_ids.append((4, analytic_salary_id.id))
+                elif slip.contract_id.analytic_account_id:
+                    move_line_analytic_ids.append(
+                        (4, slip.contract_id.analytic_account_id.id)
+                    )
```

Comment:
Please cherry pick https://github.com/OCA/payroll/pull/143

Reply:
+1

Reply:
> Please cherry pick #143

I tested ,on the side, after applying those proposals an all seems good and it is Functional OK.

Hopefully they are applied soon so this can be reviewed and approved.